### PR TITLE
pdftoedn v0.32.0

### PR DESCRIPTION
**Pull request by edporras**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Initial public release of a PDF to EDN conversion tool. 

Formerly an internal ruby gem (w/ c++ extension), we have switched to a standalone binary for greater efficiency and speed. We are using this tool on runs to process tens of thousands of documents with Starcluster. I started working on it in 2012 and it is the reason I've been submitting formula updates for new releases of poppler and leptonica.

I understand you frown upon authors submitting their own non-popular work but we've open-sourced this tool because we feel people have been looking for something like it for some time. Most existing PDF processing tools do direct conversion to a specific output format. This tool allows extraction and storage of document data which can be opened via any language with an EDN reader to extract specific document elements. For example, you can get all text spans with font and position information for each character allowing assembly of the entire document's text content for splitting and/or merging with related external documents. We also have an internal `edntosvg` tool that reproduces output that matches the source PDF amazingly well.

I'd appreciate any suggestions to get the Formula accepted. Thank you for your consideration.

### CODE REVIEW

File: Formula/pdftoedn.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Pdftoedn < Formula
+  desc "Extract PDF document data and save the output in EDN format"
+  homepage "https://github.com/edporras/pdftoedn"
+  url "https://github.com/edporras/pdftoedn/archive/v0.32.0.tar.gz"
+  sha256 "90406cb6954d01514dcea2fb6e857437afd3e8d067b81a2362a7ef6e383919bc"
+
+  bottle do
+    cellar :any
+  end
```
Comment (MikeMcQuaid) [inline Formula/pdftoedn.rb]: Please remove this; it'll be generated at build time.

File: Formula/pdftoedn.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  depends_on "boost"
+  depends_on "leptonica"
+  depends_on "openssl"
+  depends_on "rapidjson"
+
+  def install
+    ENV.cxx11 if MacOS.version < :mavericks
+
+    system "autoreconf", "-i"
+    system "./configure", "--with-openssl=/usr/local/opt/openssl/", "--prefix=#{prefix}"
```
Comment (MikeMcQuaid) [inline Formula/pdftoedn.rb]: `--with-openssl=#{Formula["openssl"].opt_prefix}`

File: Formula/pdftoedn.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  depends_on "freetype"
+  depends_on "libpng"
+  depends_on "poppler"
+  depends_on "boost"
+  depends_on "leptonica"
+  depends_on "openssl"
+  depends_on "rapidjson"
+
+  def install
+    ENV.cxx11 if MacOS.version < :mavericks
```
Comment (MikeMcQuaid) [inline Formula/pdftoedn.rb]: Feels like you should probably set this everywhere rather than just before mavericks?

File: Formula/pdftoedn.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    system "autoreconf", "-i"
+    system "./configure", "--with-openssl=/usr/local/opt/openssl/", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    # working on converting my ruby-based test suite but all the docs
+    # are private. Need to find a public set I can use.
+    system "#{bin}/pdftoedn", "-h"
```
Comment (MikeMcQuaid) [inline Formula/pdftoedn.rb]: Run something on `test_fixtures("test.pdf")` instead.

**Comment by edporras** [comment]
Hi Mike, I've made corrections as per your feedback. I also went ahead an added the first version of the test suite via `autotools`. For now there are three tests to check a few argument options and one more that processes two bundled documents and compares them against reference output I've also included. Let me know if you think anything else should be changed.

Thanks so much.

**Comment by MikeMcQuaid** [comment]
> I also went ahead an added the first version of the test suite via autotools.

That won't work; `test do` an integration test of the installed artifacts. Please make a test like I suggested earlier. Thanks!

**Comment by edporras** [comment]
Done. Sorry about that.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `pdftoedn 0.32.1 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md). Thanks!

**Comment by edporras** [comment]
Thank you, Mike. 🎉

**Pull request closed without merging**